lib/mongoid/criteria/queryable/pipeline.rb

```diff
@@ -72,16 +72,26 @@ def project(entry)
         #
         # @example Add the unwind.
         #   pipeline.unwind(:field)
+        #   pipeline.unwind(document)
+        #
+        # @see https://docs.mongodb.com/manual/reference/operator/aggregation/unwind/
         #
         # @param [ String, Symbol ] field The name of the field.
+        # @param [ Hash ]           document.
         #
         # @return [ Pipeline ] The pipeline.
         #
         # @since 2.0.0
         def unwind(field)
-          normalized = field.to_s
-          name = aliases[normalized] || normalized
-          push("$unwind" => name.__mongo_expression__)
+          unless field.respond_to? :keys
+            normalized = field.to_s
+            name = aliases[normalized] || normalized
+            push("$unwind" => name.__mongo_expression__)
+          else
```

Comment:
There's no need to check the keys of the document argument. The MongoDB server may change the names of the keys for $unwind in a future version or add other keys. We can just expect the user will provide a valid document.

In other words, this can just be ``push("$unwind" => field)``